app/src/main/java/agency/five/codebase/android/movieapp/mock/FavoritesDBMock.kt

```diff
@@ -0,0 +1,16 @@
+package agency.five.codebase.android.movieapp.mock
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+
+object FavoritesDBMock {
+
+    private val _favoriteIds = MutableStateFlow(setOf<Int>())
+    val favoriteIds: StateFlow<Set<Int>> = _favoriteIds.asStateFlow()
+
+    fun insert(movieId: Int){  _favoriteIds.value += movieId}
+    fun delete(movieId: Int){  _favoriteIds.value -= movieId}
```

Comment:
Use `_favoriteIds.update { it + movieId }` here instead of changing value for `_favoriteIds`.